scripts/top-up-cycles-ledger-account

```diff
@@ -1,38 +1,93 @@
 #!/usr/bin/env bash
 set -euo pipefail
 
+SOURCE_DIR="$(dirname "$(realpath "${BASH_SOURCE[0]}")")"
+
 print_help() {
   cat <<-EOF
-	Adds cyles to a given principal's main account on the local cycles ledger.
 
-	Usage: $(basename "$0") <principal|canister_name> <cycles>
-	E.g.   $(basename "$0") bd3sg-teaaa-aaaaa-qaaba-cai 1000000000
-	Or     $(basename "$0") my_canister 1000000000
+	Adds cyles to a given principal's main account on the cycles ledger.
 
 	Note: The destination account will be topped up by the desired amount minus fees.
+
+	Note: If you have insufficient cycles on mainnet:
+	* Check that you have ICP:
+	  dfx ledger balance --ic
+	* If not, send ICP to this account:
+	  dfx ledger --ic account-id
+	* Convert some ICP to cycles
+	  dfx cycles --ic convert --icp 1
+	* See how many cycles you have now:
+	  dfx cycles --ic balance
 	EOF
 }
 
-[[ "${1:-}" != "--help" ]] || {
+# Source the clap.bash file ---------------------------------------------------
+source "$SOURCE_DIR/clap.bash"
+# Define options
+clap.define short=n long=network desc="dfx network to use" variable=NETWORK default="local"
+clap.define short=c long=cycles desc="Number of cycles" variable=CYCLES default="1T"
+clap.define short=t long=to desc="The destinantion principal or canister name" variable=ACCOUNT default="backend"
+# Source the output file ----------------------------------------------------------
+source "$(clap.build)"
+
+: Verifies that the account has been specified
+[[ "${ACCOUNT}" ]] || {
+  echo "ERROR: Please supply the destination principal or canister name."
   print_help
-  exit 0
 }
 
-account="$1"
-sum="$2"
-account="$(dfx canister id "$account")" # Converts names to canister IDs
-echo "Balance before: $(dfx canister call cycles_ledger icrc1_balance_of '(record { owner = principal "'"$account"'"; subaccount = null })')"
-dfx wallet send "$(dfx canister id cycles_depositor)" "$sum"
-dfx canister call cycles_depositor deposit '
-(
-  record {
-    to = record {
-    owner = principal "'"$account"'";
-      subaccount = null;
-    };
-    memo = null;
-    cycles = '"$sum"' : nat;
-  },
-)
-'
-echo "Balance after: $(dfx canister call cycles_ledger icrc1_balance_of '(record { owner = principal "'"$account"'"; subaccount = null })')"
+: Converts e.g. 1T cycles to a number:
+CYCLES="${CYCLES//k/K}"
+CYCLES="${CYCLES//m/M}"
+CYCLES="${CYCLES//g/G}"
+CYCLES="${CYCLES//t/T}"
+CYCLES="${CYCLES//K/_000}"
+CYCLES="${CYCLES//M/_000_000}"
+CYCLES="${CYCLES//G/_000_000_000}"
+CYCLES="${CYCLES//T/_000_000_000_000}"
```

Comment:
shall there be a check for unexpected values here? or will it auto-fail?

Reply:
It will auto-fail.